### PR TITLE
Initial refactor and rewrite of clojure.java.jdbc docs

### PR DESCRIPTION
**Pull request by ehashman**
Addresses #229.

I've not yet had the chance to work on "Using SQL" and "Using DDL". I'll save those changes for a follow-up PR.

### CODE REVIEW

**Comment by michaelklishin** [comment]
Wow, this is awesome. Thank you!

@seancorfield can I get your blessing? :)

File: articles/ecosystem/java_jdbc/home.md [hunk 1]
```diff
@@ -3,139 +3,192 @@ title: "Using java.jdbc"
...
 
-## Setting Up A Data Source
+## Higher-level DSL and migration libraries
 
-A "database spec" is a Clojure map that specifies how to access the data source. Most commonly, you would specify the driver class name, the subprotocol, the hostname, port and database name (the "subname") and the username and password, e.g.,
+If you need more abstraction than the java.jdbc wrapper provides, you may want
+to consider using a library that provides a DSL. All of the following libraries
+are built on top of java.jdbc and provide such abstraction:
 
-    (def db-spec
```
Comment (seancorfield) [inline articles/ecosystem/java_jdbc/home.md]: I'm trying to encourage folks to switch to the new, simpler dbtype/dbname format. Could you update the example to this:
```
(def db-spec
  {:dbtype "mysql"
   :dbname "mydb"
   :user "myaccount"
   :password "secret"})
```
(and the paragraph above will need updating to reflect this preferred approach).

**Comment by seancorfield** [comment]
It's a huge improvement on what's currently there so BIG "Thanks!" to @ehashman -- I have no other comments beyond switching the introductory `db-spec` to the new, simpler format.

**Comment by ehashman** [comment]
Incoming fix for the db-spec, please hold...

**Comment by seancorfield** [comment]
Excellent! @michaelklishin Do you want to trigger a build/deploy of the site?

**Comment by michaelklishin** [comment]
Done.

**Pull request merged**